src/decoders/ascii85_decoder.rs

```diff
@@ -0,0 +1,176 @@
+///! Decode a ascii85 string
+///! Performs error handling and returns a string
+///! Call ascii85_decoder.crack to use. It returns option<String> and check with
+///! `result.is_some()` to see if it returned okay.
+///
+use crate::checkers::CheckerTypes;
+use crate::decoders::interface::check_string_success;
+use ascii85;
+
+use super::crack_results::CrackResult;
+use super::interface::Crack;
+use super::interface::Decoder;
+
+use log::{debug, info, trace};
+
+/// The ASCII85 decoder, call:
+/// `let ascii85_decoder = Decoder::<ASCII85Decoder>::new()` to create a new instance
+/// And then call:
+/// `result = ascii85_decoder.crack(input)` to decode a ascii85 string
+/// The struct generated by new() comes from interface.rs
+/// ```
+/// use ares::decoders::ascii85_decoder::{ASCII85Decoder};
+/// use ares::decoders::interface::{Crack, Decoder};
+/// use ares::checkers::{athena::Athena, CheckerTypes, checker_type::{Check, Checker}};
+///
+/// let decode_ascii85 = Decoder::<ASCII85Decoder>::new();
+/// let athena_checker = Checker::<Athena>::new();
+/// let checker = CheckerTypes::CheckAthena(athena_checker);
+///
+/// let result = decode_ascii85.crack("BOu!rD]j7BEbo7", &checker).unencrypted_text;
+/// assert!(result.is_some());
+/// assert_eq!(result.unwrap()[0], "hello world");
+/// ```
+pub struct ASCII85Decoder;
+
+impl Crack for Decoder<ASCII85Decoder> {
+    fn new() -> Decoder<ASCII85Decoder> {
+        Decoder {
+            name: "ASCII85",
+            description: "Ascii85, also called Base85, is a form of binary-to-text encoding that uses five ASCII characters to represent four bytes of binary data.",
```

Comment:
>A simple crate for encoding and decoding Adobe Ascii85

Since this is for Adobe's base85 variant (ASCII85), can you change the description and link to reflect that?

>Adobe adopted the basic btoa encoding, but with slight changes, and gave it the name Ascii85. The characters used are the ASCII characters 33 (!) through 117 (u) inclusive (to represent the base-85 digits 0 through 84), together with the letter z (as a special case to represent a 32-bit 0 value), and white space is ignored.

https://en.wikipedia.org/wiki/Ascii85#Adobe_version

Reply:
>  Adobe adopted the basic btoa encoding, but with slight changes, and gave it the name Ascii85. The characters used are the ASCII characters 33 (!) through 117 (u) inclusive (to represent the base-85 digits 0 through 84), together with the letter z (as a special case to represent a 32-bit 0 value), and white space is ignored.

All of this ^ or just the first part?

Reply:
@SkeletalDemise 👆?

Reply:
All of it! :)